scripts/components/HomePage.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from "react";
+
+
+export default class HomePage extends Component {
+  constructor(props) {
+    super(props);
+    const {server, bucket} = this.props;
+    this.state = {server, bucket};
+  }
+
+  createPoll() {
+    this.props.createPoll(this.state.server, this.state.bucket);
+  }
+
+  onChange(field, event) {
+    this.setState({[field]: event.target.value});
+  }
+
+  render() {
+    return (
+      <div id="homepage">
+        <h1>Happiness - Satisfaction tracking</h1>
+        <p>Simple polls to track satisfaction over time.</p>
+        <button type="submit" onClick={this.createPoll.bind(this)}>Create new!</button>
+        <fieldset>
+          <legend>Advanced options</legend>
+          <label>Server <input type="text" value={this.state.server} onChange={this.onChange.bind(this, "server")}/></label>
+          <label>Bucket <input type="text" value={this.state.bucket} onChange={this.onChange.bind(this, "bucket")}/></label>
```

Comment:
Perf trick in case someday you want to implement `shouldComponentUpdate`: don't bind at render time as it will create functions with different identities so you won't be able to check they're actually equal. Instead, tou can leverage class properties, like here https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/fields/ObjectField.js#L79-L83

That way you could reference `this.onChange` and alway retrieve the same reference to that function, which satisfies simple checks like `this.onChange === this.onChange`.